vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -635,6 +646,238 @@ def collect_events(self):
+                    'guest.net.ipConfig.address',
+                    1,
+                    base_tags,
+                    hostname,
+                    resource_metric_suffix,
+                    additional_tags=ip_tags,
+                )
+
+    def submit_ip_stack_property_metrics(
+        self,
+        ip_stacks,  # type: List[VmomiObject]
+        base_tags,  # type: List[str]
+        hostname,  # type: str
+        resource_metric_suffix,  # type: str
+    ):
+        # type: (...) -> None
+        for ip_stack in ip_stacks:
+            ip_tags = {}
+            if ip_stack.dnsConfig is not None:
+                host_name = ip_stack.dnsConfig.hostName
+                domain_name = ip_stack.dnsConfig.domainName
+                ip_tags.update({'route_hostname': host_name, 'route_domain_name': domain_name})
+            ip_routes = ip_stack.ipRouteConfig.ipRoute
+            for ip_route in ip_routes:
+                prefix_length = ip_route.prefixLength
+                gateway_address = ip_route.gateway.ipAddress
+                network = ip_route.network
+                # network
+                device = ip_route.gateway.device
+                route_tags = {
+                    'device': device,
+                    'network_dest_ip': network,
+                    'prefix_length': prefix_length,
+                    'gateway_address': gateway_address,
+                }
+                ip_tags.update(route_tags)
+
+                self.submit_property_metric(
+                    'guest.ipStack.ipRoute',
+                    1,
+                    base_tags,
+                    hostname,
+                    resource_metric_suffix,
+                    additional_tags=ip_tags,
+                )
+
+    def submit_basic_property_metrics(
+        self,
+        all_properties,  # type: Dict[str, Any]
+        base_tags,  # type: List[str]
+        hostname,  # type: str
+        resource_metric_suffix,  # type: str
+    ):
+        # type: (...) -> None
+        for property_name in VM_SIMPLE_PROPERTIES:
+            property_val = all_properties.get(property_name, None)
+
+            self.submit_property_metric(
+                property_name,
+                property_val,
+                base_tags,
+                hostname,
+                resource_metric_suffix,
+            )
+
+    def submit_property_metrics(
+        self,
+        resource_type,  # type: Type[vim.ManagedEntity]
+        mor_props,  # type: Dict[str, Any]
+        resource_tags,  # type: List[str]
+    ):
+        # type: (...) -> None
+        if resource_type == vim.VirtualMachine:
+            resource_metric_suffix = MOR_TYPE_AS_STRING[resource_type]
+            mor_name = to_string(mor_props.get("name", "unknown"))
+            all_properties = mor_props.get('properties', None)
+            if not all_properties:
+                self.log.warning("Could not retrieve properties for VM %s", mor_name)
```

Comment:
Are we showing this message on every check even though the properties have been deleted by the first check that reported them after the refresh? If so, couldn't it be confusing?

Reply:
We only call this method when the cache is expired so we will only log the message on check runs where we are collecting properties.